src/axom/slic/interface/slic_macros.hpp

```diff
@@ -492,14 +492,57 @@
 /// @}
 
 // Use assert when on device (HIP does not yet support assert())
-#elif defined(AXOM_DEBUG) && defined(AXOM_DEVICE_CODE) && \
-  !defined(__HIP_DEVICE_COMPILE__)
-
-  #define SLIC_ASSERT(EXP) assert(EXP)
-  #define SLIC_ASSERT_MSG(EXP, msg) assert(EXP)
-  #define SLIC_CHECK(EXP) assert(EXP)
-  #define SLIC_CHECK_MSG(EXP, msg) assert(EXP)
-
+#elif defined(AXOM_DEBUG) && defined(AXOM_DEVICE_CODE)
+  #if !defined(__HIP_DEVICE_COMPILE__)
+    #define SLIC_ASSERT(EXP) assert(EXP)
+    #define SLIC_ASSERT_MSG(EXP, msg) assert(EXP)
+    #define SLIC_CHECK(EXP) assert(EXP)
+    #define SLIC_CHECK_MSG(EXP, msg) assert(EXP)
+  #else
+// AXOM_DEBUG is defined so the expression and message cannot be ignored
+// as they probably contain parameters that were decorated with AXOM_UNUSED_PARAM
+// which will require the expression and message to be used to avoid warnings.
+// Here we use them in a way that will not really have a runtime effect.
+// The msg code block may contain << operators so we need a stream-like object
+// but we enclose it in "if(false)" so it never executes.
```

Comment:
Interesting solution! 

I suppose we should revisit if/when HIP gets an `assert()` function